contracts/registrar/ZNSSubRegistrar.sol

```diff
@@ -148,19 +143,65 @@ contract ZNSSubRegistrar is AAccessControlled, ARegistryWired, UUPSUpgradeable,
-        if (address(distrConfig.pricerContract) != address(0)) {
-            setDistributionConfigForDomain(coreRegisterArgs.domainHash, distrConfig);
+        if (address(registration.distributionConfig.pricerContract) != address(0)) {
+            setDistributionConfigForDomain(coreRegisterArgs.domainHash, registration.distributionConfig);
         }
 
         return domainHash;
     }
 
+    /**
+     * @notice Allows registering multiple subdomains in a single transaction.
+     * This function iterates through an array of `SubdomainRegistrationParams` objects and registers each subdomain
+     * by calling the `registerSubdomain` function for each entry.
+     * @dev This function reduces the number of transactions required to register multiple subdomains,
+     * saving gas and improving efficiency. Each subdomain registration is processed sequentially.
+     * 
+     * ! IMPORTANT: If a subdomain in the `subRegistrations` array has `parentHash = 0x000...` (null hash),
+     * it will be treated as a nested domain.
+     * In this case, the parent of the subdomain will be set to the domain hash of the
+     * previously registered subdomain in the array. This allows creating multi-level nested domains in a single
+     * transaction. For example:
+     * - The first subdomain must have a valid `parentHash`.
+     * - The second subdomain can have `parentHash = 0x000...`, which means it will be nested under the first subdomain.
+     * - This pattern can continue for deeper levels of nesting.
+     * 
+     * @param subRegistrations An array of `SubdomainRegistrationParams` structs, each containing:
+     *      + `parentHash`: The hash of the parent domain under which the subdomain is being registered.
+     *                     If set to `0x000...`, the parent will be the previously registered subdomain.
+     *      + `label`: The label of the subdomain to register (e.g., in `0://parent.child`, the label is `child`).
+     *      + `domainAddress`: The address to associate with the subdomain in the resolver.
+     *      + `tokenURI`: The URI to assign to the subdomain token.
+     *      + `distributionConfig`: The distribution configuration for the subdomain.
+     *      + `paymentConfig`: The payment configuration for the subdomain.
+     * @return domainHashes An array of `bytes32` hashes representing the registered subdomains.
+     */
+    function registerMultipleSubdomains(
+        SubdomainRegistrationParams[] memory subRegistrations
+    ) external override returns (bytes32[] memory) {
+
+        bytes32[] memory domainHashes = new bytes32[](subRegistrations.length);
+
+        for (uint256 i = 0; i < subRegistrations.length; i++) {
+            if (subRegistrations[i].parentHash == bytes32(0) && i > 0) {
+                subRegistrations[i].parentHash = domainHashes[i - 1];
+
+            } else if (subRegistrations[i].parentHash == bytes32(0)) {
```

Comment:
Try to always add code that reverts as early as possible in the function to save gas, having the opposite of the above check would make this easier and then switch the order